packages/service-utils/src/core/usageV2.ts

```diff
@@ -0,0 +1,56 @@
+export interface UsageV2Event {
+  /**
+   * A unique identifier for the event. Defaults to a random UUID.
+   * Useful if your service retries sending events.
+   */
+  id?: `${string}-${string}-${string}-${string}-${string}`;
```

Comment:
might be too strict on the type here, just `string` should be fine - unless you want to specifically enforce uuidv4

Reply:
It's UUID in the ClickHouse schema and I'd prefer event identifiers be consistent for all callers.